ros_buildfarm/common.py

```diff
@@ -720,6 +720,7 @@ def filter_buildfile_packages_recursively(package_names, buildfile, rosdistro_na
     implicitly_ignored = get_implicitly_ignored_package_names(
         cached_pkgs, buildfile.package_blacklist)
     res -= implicitly_ignored
+    res.difference_update(buildfile.package_ignore_list)
```

Comment:
Is there a difference (pun intended) between `-=` and `.difference_update` and a preference for using either over the other?

Reply:
There is, actually. `-=` will only accept a `set` on the RHS, where `.difference_update` will accept other collections, namely lists.

Reply:
Understood. It stands out to me to see both used adjacently but I don't think it's worth changing either or creating a set from the list just to use `-=`